bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/robot/RRMapDraw.java

```diff
@@ -346,6 +352,27 @@ private void drawOpenHabRocks(Graphics2D g2d, int width, int height, float scale
         g2d.drawString(message, textPos, height - offset * scale - stringHeight / 2);
     }
 
+    private @Nullable String getAvailableFont(String[] preferedFonts) {
+        final GraphicsEnvironment gEv = GraphicsEnvironment.getLocalGraphicsEnvironment();
+        if (gEv == null) {
+            return null;
+        }
+        String[] fonts = gEv.getAvailableFontFamilyNames();
+        if (fonts.length == 0) {
+            return null;
+        }
+        for (int j = 0; j < preferedFonts.length; j++) {
+            for (int i = 0; i < fonts.length; i++) {
+                if (fonts[i].equalsIgnoreCase(preferedFonts[j])) {
+                    return preferedFonts[j];
+                }
+            }
+        }
+        // Preferred fonts not available... do a lucky draw
+        Random r = new Random();
+        return fonts[r.nextInt(fonts.length)];
```

Comment:
While it is amusing, doing this would cause great confusion to users that got random fonts since it would change every time the map is drawn. Instead just go for the first font on the list and hope it isn't wingdings.
```suggestion
        // Preferred fonts not available... just go with the first one
        return fonts[0];
```